modules/covid_vaccine/app/services/covid_vaccine/v0/registration_service.rb

```diff
@@ -99,10 +99,22 @@ def attributes_from_mpi(form_data)
         end
       end
 
-      # if user_type == loa3, then we already had the information from their
-      # authenticated session and added it to the raw form data
+      ## Guard around MPI query
+      # 1. If user_type == loa3, we already have their information from MPI in the
+      # authenticated session
+      # 2. If not all of the required MPI query keys are present, we can't query MPI
+      # 3. If a partial, unparseable date of birth was submitted, we can't query MPI
+      #
       def should_query_mpi?(form_data, user_type)
-        user_type != 'loa3' && (REQUIRED_QUERY_TRAITS & form_data.keys).size == REQUIRED_QUERY_TRAITS.size
+        return false if user_type == 'loa3'
+        return false unless (REQUIRED_QUERY_TRAITS & form_data.keys).size == REQUIRED_QUERY_TRAITS.size
```

Comment:
hmm, i don't understand why we need this. Wouldn't it be better do: 

`form_data.select { |h| h.key.in?(REQUIRED_QUERY_TRAITS) && h.value.present? }` 

or something instead?

Reply:
you're comparing the intersection of of those keys to make sure they're all present so i guess that's a little bit different than what I'm proposing. Nevermind, but I would think a presence check is still needed, because the key could be present but value would be nil.

Reply:
your expression would find all the supplied attributes that were required, but it wouldn't ensure that *all* required elements were present. I'm doing an intersection and comparing size to check that all required keys are present. You're right that I don't check for a non-empty values though...

Reply:
`(REQUIRED_QUERY_TRAITS & form_data.select { |h| h.value.present? }.keys).size == REQUIRED_QUERY_TRAITS.size`

maybe something like ☝️

Reply:
resolving per comment on slack:

> i think so. Only downside is not checking for blank parameter values but all that will do is result in a mpi query with no results